modules/gapi/include/opencv2/gapi/own/rmat.hpp

```diff
@@ -0,0 +1,52 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2020 Intel Corporation
+
+
+#ifndef OPENCV_GAPI_OWN_RMAT_HPP
+#define OPENCV_GAPI_OWN_RMAT_HPP
+
+#include <opencv2/gapi/gmat.hpp>
+
+namespace cv { namespace gapi { namespace own {
+
+class HostMat
+{
+public:
+    virtual ~HostMat() = default;
+    virtual Mat getMat() const = 0;
+    virtual void flush() const = 0;
+};
```

Comment:
What this `HostMat` stands for and should we really expose it?

Reply:
`HostMat` is a wrapper of a memory which is mapped to a host and can be accessed by the host. All this methods can be moved to `RMat` not to introduce new entities. Done it this way to make `RMat` as lightweight as possible: if we don't need to map memory to host and just want to describe some workload on some remote memory, we can do it without `HostMat` at all and use only `RMat`s current interface